app/hypnos/src/gui.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020 Endian Technologies AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/sensor/cst816s.h>
+#include "gfx.h"
+#include "clock.h"
+#include "battery.h"
+#include "bt.h"
+#include "log.h"
+
+static enum cst816s_gesture gesture;
+static struct sensor_value touch_point;
+static enum screen sc;
+
+void gui_handle_touch_event(struct device *touch_dev)
+{
+	if (sensor_sample_fetch(touch_dev) < 0) {
+		LOG_ERR("Touch sample update error.");
+	}
+	struct sensor_value gesture_val;
+	sensor_channel_get(touch_dev, CST816S_CHAN_GESTURE, &gesture_val);
+	gesture = gesture_val.val1;
+
+	sensor_channel_get(touch_dev, CST816S_CHAN_TOUCH_POINT_1, &touch_point);
+	LOG_INF("Gesture %d on x=%d, y=%d", gesture, touch_point.val1, touch_point.val2);
```

Comment:
Maybe keep the sample fetch and chaneel get in the ISR ? and give the gesture as a parameter to this function.

Reply:
fixed